Dockerfile

```diff
@@ -1,75 +1,76 @@
-		\
-		# Archived versions 1.7 and under use some absolute links, and v1.10 uses
-		# "relative" links to sources (href="./css/"). Remove those to make them
-		# work :)
-		&& find target/${VER} -type f -name '*.html' -print0 | xargs -0 sed -i 's#href="\./#href="/#g' \
-		&& find target/${VER} -type f -name '*.html' -print0 | xargs -0 sed -i 's#src="\./#src="/#g' \
-		\
-		# Create permalinks for archived versions
-		\
-		&& find target/${VER} -type f -name '*.html' -print0 | xargs -0 sed -i 's#href="/#href="/'"$BASEURL"'#g' \
-		&& find target/${VER} -type f -name '*.html' -print0 | xargs -0 sed -i 's#src="/#src="/'"$BASEURL"'#g'; \
-  done; \
-  rm -rf archive_source
+# Get archival docs from each canonical image
+FROM docs/docker.github.io:v1.4 AS archive_v1.4
+FROM docs/docker.github.io:v1.5 AS archive_v1.5
+FROM docs/docker.github.io:v1.6 AS archive_v1.6
+FROM docs/docker.github.io:v1.7 AS archive_v1.7
+FROM docs/docker.github.io:v1.8 AS archive_v1.8
+FROM docs/docker.github.io:v1.9 AS archive_v1.9
+FROM docs/docker.github.io:v1.10 AS archive_v1.10
+FROM docs/docker.github.io:v1.11 AS archive_v1.11
+FROM docs/docker.github.io:v1.12 AS archive_v1.12
+FROM docs/docker.github.io:v1.13 AS archive_v1.13
+FROM docs/docker.github.io:v17.03 AS archive_v17.03
+FROM docs/docker.github.io:v17.06 AS archive_v17.06
+
+# Reset with nginx, so we don't get docs source in the image
+FROM nginx:alpine AS docs_base
+
+COPY _scripts/fix_archives.sh /usr/bin/fix_archives.sh
+
+ENV TARGET=/etc/nginx/html/
+
+# Copy HTML from each stage above and run a script to fix relative links
+
+COPY --from=archive_v1.4 /etc/nginx/html ${TARGET}/v1.4
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v1.4'
+
+COPY --from=archive_v1.5 /etc/nginx/html ${TARGET}/v1.5
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v1.5'
+
+COPY --from=archive_v1.6 /etc/nginx/html ${TARGET}/v1.6
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v1.6'
+
+COPY --from=archive_v1.7 /etc/nginx/html ${TARGET}/v1.7
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v1.7'
+
+COPY --from=archive_v1.8 /etc/nginx/html /${TARGET}/v1.8
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v1.8'
+
+COPY --from=archive_v1.9 /etc/nginx/html ${TARGET}/v1.9
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v1.9'
+
+COPY --from=archive_v1.10 /etc/nginx/html ${TARGET}/v1.10
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v1.10'
+
+COPY --from=archive_v1.11 /etc/nginx/html ${TARGET}/v1.11
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v1.11'
+
+COPY --from=archive_v1.12 /etc/nginx/html ${TARGET}/v1.12
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v1.12'
+
+COPY --from=archive_v1.13 /etc/nginx/html ${TARGET}/v1.13
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v1.13'
+
+COPY --from=archive_v17.03 /etc/nginx/html ${TARGET}/v17.03
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v17.03'
+
+COPY --from=archive_v17.06 /etc/nginx/html ${TARGET}/v17.06
+RUN bash /usr/bin/fix_archives.sh ${TARGET} 'v17.06'
```

Comment:
Ok for a follow-up, but what do you think about adding the _current_ documentation as an "archive" as well? This allows using perma-links to sections in the documentation for the current release

Reply:
It's an interesting idea for later, but could wreck SEO if we did it wrong.

Reply:
We may need [`canonical` headers](https://moz.com/blog/canonical-url-tag-the-most-important-advancement-in-seo-practices-since-sitemaps) to prevent the duplicate content penalty (same for all archives)